osu.Game.Tests/Visual/Components/TestScenePreviewTrackManager.cs

```diff
@@ -108,6 +108,26 @@ public void TestNonPresentTrack()
             AddAssert("track stopped", () => !track.IsRunning);
         }
 
+        /// <summary>
+        /// Ensures that <see cref="PreviewTrackManager.CurrentTrack"/> changes correctly.
+        /// </summary>
+        [Test]
+        public void TestCurrentTrackChanges()
+        {
+            PreviewTrack track = null;
+            TestTrackOwner owner = null;
+
+            AddStep("get track", () => Add(owner = new TestTrackOwner(track = getTrack())));
+            AddUntilStep("wait loaded", () => track.IsLoaded);
+            AddStep("start track", () => track.Start());
+            AddAssert("current is track", () => trackManager.CurrentTrack == track);
+            AddStep("pause manager updates", () => trackManager.AllowUpdate = false);
+            AddStep("stop any playing", () => trackManager.StopAnyPlaying(owner));
+            AddAssert("current not changed", () => trackManager.CurrentTrack == track);
+            AddStep("resume manager updates", () => trackManager.AllowUpdate = true);
+            AddAssert("current is null", () => trackManager.CurrentTrack == null);
```

Comment:
how is this testing that mute was undone? and why is the "allowupdate" required?

Reply:
well, this test is ensuring that the current track changes correctly as we're assuming `Stop` will remove the audio bindable, `AllowUpdate` is required to check if the `CurrentTrack` didn't change before the scheduled stopped event is invoked.

Reply:
why not test for it? that's generally what we do.

Reply:
In general `current == null` is the less important half of the desired behaviour we want here. With regression tests you generally want to test the actually important user-facing part of the regression, which is the volume (from what I should see it should be easily exposed). `current` being `null` could be well considered an implementation detail.